src/content/docs/en/recipes/build-custom-img-component.mdx

```diff
@@ -0,0 +1,111 @@
+---
+title: Build a custom image component 
+description: Learn how to build a custom image component that supports media queries using the getImage function
+i18nReady: true
+type: recipe
+---
+
+Astro provides a couple of built in components that you can use to display and optimize your images.  The `<Picture>` component allows you to display responsive images and work with different formats and sizes. The `<Image>` componet will optimze your images and allow you to pass in different formats and quality properties.
+
+But there will be times where you will need to have more customizable options for your images that the `<Picture>` and `<Image>` components currently do not support. 
+
+In this recipe, you will learn how use the `getImage()` function to create your own custom image component that supports the use of media queries/art direction.  The purpose of the `getImage()` function is to create images for use outside of direct embedding in HTML.
+
+## Recipe
+
+1. Create a new Astro component and import the `getImage()` function
+
+    ```astro title="src/components/MyCustomImageComponent.astro"
+    ---
+     import { getImage } from "astro:assets";
+    ---
+
+    ```
+
+2. Destructure the `props` from `Astro.props`. You can also use TypeScript to type the props. 
```

Comment:
```suggestion
2. Destructure three `props` from `Astro.props`: `mobileImgUrl` and `desktopImgUrl` for creating your image at different viewport sizes, as well as `alt` for the image's alt text. These props will be passed wherever you render your custom image components. You can also use TypeScript to type the props. 
```
So, what about something kind of like this? It felt a little early to say "destructure THE props" when we haven't seen anything yet? So, I'm thinking of something that introduces them a tiny bit?

Also, while destructure is totally fine, you could also consider a phrasing something like:

> Your custom component will receive/use three props..
or
> Create three props that your custom component will receive...

Again, nothing wrong with using destructure, but if the emphasis is now on introducing these props that will be created, then maybe it's not as easy to squish those ideas into a sentence starting with destructure. So, try some things, and see what you like!

Reply:
yeah, the notes about destructuring makes sense here.
I will make some updates 👍